File: Chapter_4/project_4_4.pl
use strict;
use warnings;
use Const::Fast;

const my $OCTAL_DIV => 8;
printf 'Enter a number between 0 and 32767: ';
chomp( my $user_input = (<STDIN>) );

my $five  = $user_input % $OCTAL_DIV;
my $four  = ( $user_input / $OCTAL_DIV ) % $OCTAL_DIV;
my $three = ( ( $user_input / $OCTAL_DIV ) / $OCTAL_DIV ) % $OCTAL_DIV;
my $two =
  ( ( ( $user_input / $OCTAL_DIV ) / $OCTAL_DIV ) / $OCTAL_DIV ) % $OCTAL_DIV;
my $one =
  ( ( ( ( $user_input / $OCTAL_DIV ) / $OCTAL_DIV ) / $OCTAL_DIV ) /
      $OCTAL_DIV )
  % $OCTAL_DIV;

printf "In Octal your number is %d%d%d%d%d\n", $one, $two, $three, $four, $five;

1;

File: Chapter_4/project_4_4.py
user_input = int(input("Enter a number between 0 and 32767: "))
five = user_input % 8
four = (user_input // 8) % 8
three = ((user_input // 8) // 8) % 8
two = (((user_input // 8) // 8) // 8) % 8
one = ((((user_input // 8) // 8) // 8) // 8) % 8
print(f"In Octal your number is {one}{two}{three}{four}{five}")